### PR TITLE
Fix metric resizing

### LINKED ISSUE #231: Bug in metric resizing
There seems to be a bug in the code to resize the mass matrix.

MWE:
```
using AdvancedHMC, Distributions, ForwardDiff

D = 10; initial_θ = rand(D)
ℓπ(θ) = logpdf(MvNormal(zeros(D), ones(D)), θ)
n_samples = 200

metric = DiagEuclideanMetric(1)
hamiltonian = Hamiltonian(metric, ℓπ, ForwardDiff)

proposal = NUTS{MultinomialTS, GeneralisedNoUTurn}(Leapfrog(0.1))

samples, stats = sample(hamiltonian, proposal, initial_θ, n_samples)
```

This gives me the error:
```
ERROR: LoadError: MethodError: no method matching DiagEuclideanMetric{Float64,Array{Float64,1}}(::Tuple{Int64})
Closest candidates are:
  DiagEuclideanMetric{Float64,Array{Float64,1}}(::Any, ::Any, ::Any) where {T, A<:Union{AbstractArray{T,1}, AbstractArray{T,2}}} at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/metric.jl:30
Stacktrace:
 [1] resize(::Hamiltonian{DiagEuclideanMetric{Float64,Array{Float64,1}},typeof(ℓπ),typeof(ℓπ)}, ::Array{Float64,1}) at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/sampler.jl:23
 [2] sample_init at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/sampler.jl:39 [inlined]
 [3] #sample#19(::Bool, ::Bool, ::Bool, ::typeof(AdvancedHMC.pm_next!), ::typeof(sample), ::Random._GLOBAL_RNG, ::Hamiltonian{DiagEuclideanMetric{Float64,Array{Float64,1}},typeof(ℓπ),typeof(ℓπ)}, ::NUTS{MultinomialTS,GeneralisedNoUTurn,Leapfrog{Float64},Float64}, ::Array{Float64,1}, ::Int64, ::AdvancedHMC.Adaptation.NoAdaptation, ::Int64) at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/sampler.jl:172
 [4] #sample at ./int.jl:0 [inlined]
 [5] #sample#18 at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/sampler.jl:116 [inlined]
 [6] sample at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/src/sampler.jl:116 [inlined] (repeats 2 times)
 [7] top-level scope at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/mwes/resizing/mwe.jl:12
 [8] include at ./boot.jl:328 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1105
 [10] include(::Module, ::String) at ./Base.jl:31
 [11] include(::String) at ./client.jl:424
 [12] top-level scope at REPL[2]:1
in expression starting at /Users/reichelt/dev/projects/julia/AdvancedHMC.jl/mwes/resizing/mwe.jl:12
```

I will submit a PR shortly with one possible way to fix the bug.

### PR DESCRIPTION
**Pull request by treigerm**
This is one way to fix #231, I have also added a simple test.

### CODE REVIEW

File: src/sampler.jl [hunk 1]
```diff
@@ -20,7 +20,7 @@ end
 function resize(h::Hamiltonian, θ::AbstractVecOrMat{T}) where {T<:AbstractFloat}
     metric = h.metric
     if size(metric) != size(θ)
-        metric = typeof(metric)(size(θ))
+        metric = getname(typeof(metric))(size(θ))
```
Comment (yebai) [inline src/sampler.jl:23]: this looks good overall. For clarity, maybe consider merging `getname(typeof(metric))` into `getname(metric)`, i.e. calling the `typeof` function inside `getname`?

**Review by yebai — APPROVED** [review]

**Comment by treigerm** [comment]
Thank you for the quick review! I made the suggested change.

**Comment by yebai** [comment]
Would you like to bump the version number from `v0.2.26` to `v0.2.27` so that a new bugfix release can be tagged after merging this PR? 

https://github.com/TuringLang/AdvancedHMC.jl/blob/master/Project.toml#L3

**Comment by treigerm** [comment]
Done!

**Comment by yebai** [comment]
Thanks, this should be fixed after a new release is tagged: 

https://github.com/TuringLang/AdvancedHMC.jl/commit/e00b6867a6b28baec580e67f71ae05d79342defe#commitcomment-44692011

**Pull request merged**